iati/home/templates/home/includes/sidepanel.html

```diff
@@ -7,52 +7,127 @@ <h3 class="aside-m__heading">In this section</h3>
     <button class="navigation-secondary-toggle" type="button" id="navigation-secondary-toggle">pages in this section <i role="presentation" aria-hidden="true" class="ico ico--16 ico-plus-white"></i></button>
     <nav class="navigation-secondary" id="navigation-secondary">
         <ul class="navigation-secondary__list">
-          {% if self.depth == 3 %}
+          {% if self.depth == 3 %} {# Pages with depth of 3 are child pages of this section #}
```

Comment:
I think you might have the logic of this page backwards a bit. The template needs to change depending on the depth of the page on which it appears. So pages with a depth of 3 are not child pages of this section, `self.depth == 3` tells us that this page is a child of the highest level we don't display (depth 2, e.g. About).

Reply:
I tried building the hierarchy from the top-down, but it was impossible to determine the path on which the current page existed. So this hierarchy is built from the bottom up. Here's the logic in pseudo-code:

1. If the page you're on is a default page (depth 3), show the immediate children.
2. If the page you're on is a child of a default page (depth 4), query the parent of the current page for sibling pages and query the current page for children.
3. If the page you're on is a child of a child of a default page (depth 5), query the parent of the parent of the current page for aunt and uncle pages, query your immediate parent for sibling pages, and query the current page for children.